File: AD-HOC/1104/main.rb
# frozen_string_literal:true

require 'set'

a, b = gets.split.map(&:to_i)
while a != 0 && b != 0
  a_set = gets.split.map(&:to_i).to_set
  b_set = gets.split.map(&:to_i).to_set
  union_set = a_set | b_set
  difference = [union_set.size - a_set.size, union_set.size - b_set.size].min
  puts difference
  a, b = gets.split.map(&:to_i)
end

File: AD-HOC/1104/main.js
let input = require('fs').readFileSync('/dev/stdin', 'utf-8');
let lines = input.split('\n');
let i = 0;
let [a, b] = lines[i++].split(' ').map(Number);
while (a !== 0 && b !== 0) {
    let a_set = new Set(lines[i++].split(' ').map(Number));
    let b_set = new Set(lines[i++].split(' ').map(Number));
    let union_set = new Set([...a_set, ...b_set]);
    let difference = Math.min(union_set.size - a_set.size, union_set.size - b_set.size);
    console.log(difference);
    [a, b] = lines[i++].split(' ').map(Number);
}
